Tests/MongoSwiftTests/UnifiedTestRunner/UnifiedTestRunner.swift

```diff
@@ -248,17 +254,52 @@ struct UnifiedTestRunner {
                             }
                         }
                     }
+                    // Manual clean up
+                    try await self.terminateOpenTransactions()
+                    await context.disableFailpoints()
+                    try await self.closeEntities(context: context)
                 } catch let testErr {
                     // Test runners SHOULD terminate all open transactions after each failed test by killing all
                     // sessions in the cluster.
                     do {
-                        try self.terminateOpenTransactions()
+                        // Clean up to prevent a client closing error from obscuring the real error
+                        try? await self.terminateOpenTransactions() // bug
+                        await context.disableFailpoints()
+                        try await closeEntities(context: context)
                     } catch {
                         print("Failed to terminate open transactions: \(error)")
                     }
                     throw testErr
                 }
             }
         }
+
+        try self.internalClient.closeClients()
+    }
+
+    func closeEntities(context: Context) async throws {
+        // Close streams and cursors
```

Comment:
I think it would be helpful to future expand this comment to clarify that cursors and change streams must be killed before their parent clients, hence the separate loops